src/parse_constants.rs

```diff
@@ -340,6 +340,10 @@ impl ParseError {
             .position(|c| *c == '\n')
             .map(|pos| pos + last_char_in_range)
             .unwrap_or(src.len());
+        // We can only report squiggles on one line
+        if start + len > line_end {
+            len = line_end - start;
```

Comment:
I wonder what the behavior should be for other multiline nodes.
Somehow this doesn't fix this one:

```
fish -c 'time begin
  echo
  end &'
fish: 'time' is not supported for background jobs. Consider using 'command time'.
time begin
echo
end &
^
```

Reply:
Yup, it passes the entire job_node for context: https://github.com/fish-shell/fish-shell/blob/67d78fb2582725515e8ef62a05dc0abfa228ff68/src/parse_execution.rs#L1547

It could possibly mark only the "time" part, and possibly print the other lines after the marker, like

```
time begin
^~~^
echo
end &
```

Reply:
Even if we restrict the error location to a single keyword, it could span multiple lines due to line continuations.
Would be nice if we had a simple rendering algorithm for all cases.

```
$ fish -c 'ti\
  me echo asdf &'
fish: 'time' is not supported for background jobs. Consider using 'command time'.
ti\
me echo asdf &
^
```

I guess we could highlight the offending parts (`time` and `&` here) in `fish_color_error`
(which we could also do before executing),
and not show the `^` if we think the width is -1 (but we should make it not affect the `function eval` case).